fact_extractor/helperFunctions/install.py

```diff
@@ -144,26 +132,39 @@ def install_github_project(project_path: str, commands: List[str]):
         for command in commands:
             output, return_code = execute_shell_command_get_return_code(command)
             if return_code != 0:
-                error = 'Error while processing github project {}!\n{}'.format(project_path, output)
+                error = f'Error while processing github project {project_path}!\n{command}\n{output}'
                 break
 
     if error:
         raise InstallationError(error)
 
 
 def _checkout_github_project(github_path, folder_name):
-    clone_url = 'https://www.github.com/{}'.format(github_path)
-    stdout, return_code = execute_shell_command_get_return_code('git clone {}'.format(clone_url))
+    clone_url = f'https://www.github.com/{github_path}'
+    stdout, return_code = execute_shell_command_get_return_code(f'git clone {clone_url}')
     if return_code != 0:
-        raise InstallationError('Cloning from github failed for project {}: {}\n'.format(github_path, stdout))
+        raise InstallationError(f'Cloning from github failed for project {github_path}: {stdout}\n')
     if not Path('.', folder_name).exists():
-        raise InstallationError('Repository creation failed on folder {}: {}\n'.format(folder_name, stdout))
+        raise InstallationError(f'Repository creation failed on folder {folder_name}: {stdout}\n')
 
 
 def load_main_config():
     config = configparser.ConfigParser()
     config_path = Path(Path(__file__).parent.parent, 'config', 'main.cfg')
     if not config_path.is_file():
-        raise InstallationError('Could not load config at path {}'.format(config_path))
+        raise InstallationError(f'Could not load config at path {config_path}')
     config.read(str(config_path))
     return config
+
+
+def is_virtualenv() -> bool:
+    """Check if FACT runs in a virtual environment"""
+    return sys.prefix != getattr(sys, 'base_prefix', getattr(sys, 'real_prefix', None))
+
+
+def gcc_is_new() -> bool:
+    try:
+        output = run(split('gcc --version'), text=True, capture_output=True).stdout
+        return int(output.splitlines()[0].split(' ')[-1].split('.')[0]) >= 10
+    except ValueError:
+        return False
```

Comment:
This is some black magic here, hm. Why is `10` new? Should this not be parametrized?
The function's purpose would be clearer if it would go sth. like this:

```suggestion
def check_gcc_major_version_at_least(version: int = 10) -> bool:
    try:
        output = run(split('gcc --version'), text=True, capture_output=True).stdout
        major_version = int(output.splitlines()[0].split(' ')[-1].split('.')[0])
        return  major_version >= version
    except ValueError:
        return False
```

Reply:
> Why is 10 new?
AFAIK 10 was the version where changes to default warning flags occurred (which can lead to build errors with `-Werror`)

> Should this not be parametrized?
Yes, good idea!

Reply:
I also just found out that there is a `gcc -dumpversion` option which should make this less ugly.

Reply:
FTR: switching to a [newer Fork of sasquatch](https://github.com/threadexio/sasquatch) made this unnecessary so I removed it completely